pkg/kfake/cluster.go

```diff
@@ -183,6 +186,10 @@ func (c *Cluster) run() {
 			creq = w.creq
 		case <-c.die:
 			return
+		case fn := <-c.adminCh:
+			// Run a custom request in the context of the cluster
+			fn()
```

Comment:
I like this a lot, this is the same thing I do in kgo/metadata.go: https://github.com/twmb/franz-go/blob/1a59c2d62d0d0ebd115f4f507379189644972c1f/pkg/kgo/metadata.go#L183